pkg/operator/controller/dns/controller.go

```diff
@@ -72,6 +100,26 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return reconcile.Result{}, fmt.Errorf("failed to get dns 'cluster': %v", err)
 	}
 
+	infraConfig := &configv1.Infrastructure{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get infrastructure 'config': %v", err)
+	}
+
+	// Initialize or re-initialize the DNS provider.
+	if r.infraConfig == nil || !reflect.DeepEqual(infraConfig.Spec, r.infraConfig.Spec) || !reflect.DeepEqual(infraConfig.Status, r.infraConfig.Status) {
```

Comment:
@Miciah why is it necessary to compare spec of `configv1.Infrastructure`?

Reply:
Thanks!  I've removed the ` || !reflect.DeepEqual(infraConfig.Spec, r.infraConfig.Spec)`.